src/Views/AppListUpdateView.vala

```diff
@@ -227,21 +213,14 @@ namespace AppCenter.Views {
 
             update_all_button.clicked.connect (on_update_all);
 
-            unowned var flatpak_backend = AppCenterCore.FlatpakBackend.get_default ();
-            if (!flatpak_backend.working) {
-                on_updates_changed ();
-            }
+            flatpak_backend.updatable_packages.items_changed.connect (on_updates_changed);
+            on_updates_changed ();
 
             flatpak_backend.notify ["working"].connect (() => {
                 if (flatpak_backend.working) {
                     refresh_menuitem.sensitive = false;
-                    header_revealer.reveal_child = false;
-                    updated_revealer.reveal_child = false;
```

Comment:
I think this might be a small regression. If we're in the process of checking for updates, we shouldn't be reporting that everything is already up to date